core/src/com/unciv/models/Religion.kt

```diff
@@ -40,6 +40,24 @@ class Religion() : INamed {
         this.gameInfo = gameInfo
     }
 
+    fun getPantheonBeliefs(): Sequence<Belief> {
+        val rulesetBeleifs = gameInfo.ruleSet.beliefs
+        return followerBeliefs.mapNotNull {
+            if (it !in rulesetBeleifs) null
+            else rulesetBeleifs[it]!!
+        }.filter { it.type == "Pantheon" }
+            .asSequence()
+    }
+    
+    fun getFollowerBeliefs(): Sequence<Belief> {
+        val rulesetBeliefs = gameInfo.ruleSet.beliefs
+        return followerBeliefs.mapNotNull {
+            if (it !in rulesetBeliefs) null
+            else rulesetBeliefs[it]!!
```

Comment:
Instead of doing this twice, create a getFollowerBeliefsInRuleset():Sequence<Belief> function